### PR TITLE
Fish Refinery

### PR DESCRIPTION
**Pull request by ShinyPulsar**
## About The Pull Request
Adds a fish refinery able to turn fish into raw PE.
Conversion is 20 fishes for 1 raw PE box.
Refining time is 2 minutes.
Accepts any fish.

## Why It's Good For The Game
Gives the opportunity to people to help the facility through fishing. Exclusive to fishing maps obviously so people might be interested to vote for these maps instead of constantly playing ALPHA.

## Changelog
:cl:
add: Added Fish Refinery
:cl:

### CODE REVIEW

File: ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/obj/structure/fish_refinery
+	name = "Fish Refinery"
+	desc = "A machine used to refine any fish into raw PE."
+	icon = 'ModularTegustation/fishing/icons/fishmachines.dmi'
+	icon_state = "fish_refinery"
+	anchored = TRUE
+	density = TRUE
+	resistance_flags = INDESTRUCTIBLE
+	var/refine_timer = 2 MINUTES
+	var/timeleft
```
Comment (Gboster-0) [inline ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm:10]: this variable is never used due to being a left-over from the old procs, so we can delete it
```suggestion
```

File: ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			fish_amount +=1
+			to_chat(user, span_notice("You load fish into the machine."))
+			qdel(I)
+			playsound(get_turf(src), 'sound/misc/box_deploy.ogg', 5, 0)
+		else
+			to_chat(user, span_notice("Something is already loaded."))
+		if(fish_amount ==20)
+			loaded = TRUE
+			fish_amount = 0
+			timeleft = refine_timer
```
Comment (Gboster-0) [inline ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm:43]:
```suggestion
```

File: ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			return
+
+
+/obj/structure/fish_refinery/proc/counter()
+	if(loaded==TRUE)
+		sleep(refine_timer)
+		loaded = FALSE
+		new /obj/item/rawpe(get_turf(src))
+		visible_message(span_notice("The refinery finishes refining a box."))
+	else
```
Comment (Gboster-0) [inline ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm:55]: the code auto-returns if there's no more code to execute, so this `if else` does nothing
```suggestion
```

File: ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		return
```
Comment (Gboster-0) [inline ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm:56]:
```suggestion
```

File: ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+/obj/structure/fish_refinery/Destroy()
+	GLOB.lobotomy_devices -= src
+	if(loaded)
+		new /obj/item/food/fish(get_turf(src))
+	return ..()
+
+/obj/structure/fish_refinery/examine(mob/user)
+	. = ..()
+	if(loaded==FALSE)
```
Comment (Gboster-0) [inline ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm:26]: placing an exclamation point before a variable name in an if() check is a "better" way to handle if(X == FALSE) checks
```suggestion
	if(!loaded)
```

File: ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+			loaded = TRUE
+			fish_amount = 0
+			timeleft = refine_timer
+			INVOKE_ASYNC(src, PROC_REF(counter))
+		else
+			return
+
+
+/obj/structure/fish_refinery/proc/counter()
+	if(loaded==TRUE)
```
Comment (Gboster-0) [inline ModularTegustation/fishing/code/fishing_items/fishing_refinery.dm:50]:
```suggestion
	if(loaded)
```

**Review by Gboster-0 — COMMENTED** [review]
some nitpicks

**Review by Kitsunemitsu — APPROVED** [review]
I know why you didn't make it a subtype of the alt_refinery now, I will merge this; and will look into ways to implement this later.

**Pull request merged**